apps/election-manager/src/lib/votecounting.ts

```diff
@@ -485,6 +385,171 @@ export const getContestTallyMeta = ({
   }, {})
 }
 
+function getTallyForCastVoteRecords(
+  election: Election,
+  castVoteRecords: CastVoteRecord[]
+): Tally {
+  const allVotes: VotesDict[] = []
+  const cvrMap = new Map<string, CastVoteRecord>()
+  let numberOfBallotsCounted = 0
+  castVoteRecords.forEach((CVR) => {
+    const vote = buildVoteFromCvr({ election, cvr: CVR })
+    cvrMap.set(CVR._ballotId, CVR)
+    allVotes.push(vote)
+    numberOfBallotsCounted += 1
+  })
+
+  const overallTally = tallyVotesByContest({ election, votes: allVotes })
+  const contestTallyMetadata = getContestTallyMeta({
+    election,
+    castVoteRecords,
+  })
+
+  return {
+    contestTallies: overallTally,
+    castVoteRecords: cvrMap,
+    numberOfBallotsCounted,
+    contestTallyMetadata,
+  }
+}
+
+export function computeFullElectionTally(
+  election: Election,
+  castVoteRecordLists: CastVoteRecordLists
+): OptionalFullElectionTally {
```

Comment:
This doesn't have to be optional.
```suggestion
): FullElectionTally {
```
Fixing this will remove the need for the non-null assertions you added in various places.